File: Command/command.cs
namespace ConsoleApplication
{
  class MainApp
  {
    static void Main (string[] args)
    {
      var receiver = new Receiver ();
      var command = new ConcreteCommand (receiver);
      var invoker = new Invoker ();

      invoker.Command = command;
      invoker.ExecuteCommand ();
    }
  }

  class Receiver
  {
    public void Action ()
    {
      Console.WriteLine ("Receiver.Action()");
    }
  }

  abstract class Command
  {
    protected Receiver receiver;

    public Command (Receiver receiver)
    {
      this.receiver = receiver;
    }

    public abstract void Execute ();
  }

  class ConcreteCommand : Command
  {
    public ConcreteCommand (Receiver receiver) : base(receiver)
    {
    }

    public override void Execute ()
    {
      this.receiver.Action ();
    }
  }

  class Invoker
  {
    private Command _command;

    public Command Command {
      private get { return this._command; }
      set { this._command = value; }
    }

    public void ExecuteCommand ()
    {
      this.Command.Execute ();
    }
  }
}

File: Command/command.php
<?php

class Receiver {
	public function write($str) {
		echo $str;
	}
}

class Command implements CommandInterface {
	protected $output;

	function __construct(Receiver $output) {
		$this->output = $output;
	}

	public function execute() {
		$this->output->write('Hello!');
	}
}

class Invoker {
	protected $command;

	public function setCommand(CommandInterface $cmd ) {
		$this->command = $cmd;
	}

	public function run() {
		$this->command->execute();
	}
}

interface CommandInterface {
	public function execute();
}

$receiver = new Receiver();
$command  = new Command($receiver);
$invoker  = new Invoker();

$invoker->setCommand($command);
$invoker->run();
